cobbler/items/abstract/inheritable_item.py

```diff
@@ -194,21 +194,27 @@ def parent(self) -> Optional[Union["System", "Profile", "Distro", "Menu"]]:
         return self.api.get_items(self.COLLECTION_TYPE).get(self._parent)  # type: ignore
 
     @parent.setter
-    def parent(self, parent: str) -> None:
+    def parent(self, parent: Union["InheritableItem", str]) -> None:
```

Comment:
Why is this change needed? When you have an item at hand you can also just directly pass in the name instead of using the full item.

Reply:
Without this it is not possible to set unsaved item as parent, so parent-child can't be created in one transaction.